workers/unity/Packages/com.improbable.gdk.tools/RedirectedProcess.cs

```diff
@@ -266,41 +258,121 @@ public int Run()
             }
         }
 
-        /// <summary>
-        ///     Runs the redirected process and returns a task which can be waited on.
-        /// </summary>
-        /// <returns>A task which would return the exit code and output.</returns>
-        public Task<RedirectedProcessResult> RunAsync()
+        private (Process, StringBuilder) SetupProcess()
         {
-            return Task.Run(() =>
+            var info = new ProcessStartInfo(command, string.Join(" ", arguments))
             {
-                var processStandardOutput = new List<string>();
-                var processStandardError = new List<string>();
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                WorkingDirectory = workingDirectory
+            };
 
-                AddOutputProcessing(output =>
+            var process = new Process
+            {
+                StartInfo = info
+            };
+
+            StringBuilder outputLog = null;
+            if ((outputRedirectBehaviour & OutputRedirectBehaviour.RedirectAccumulatedOutput) !=
+                OutputRedirectBehaviour.None)
+            {
+                outputLog = new StringBuilder();
+            }
+
+            process.OutputDataReceived += (sender, args) =>
+            {
+                var outputString = args.Data;
+                if (string.IsNullOrEmpty(outputString))
                 {
-                    lock (processStandardOutput)
+                    return;
+                }
+
+                if ((outputRedirectBehaviour & OutputRedirectBehaviour.ProcessSpatialOutput) != OutputRedirectBehaviour.None)
+                {
+                    outputString = ProcessSpatialOutput(outputString);
+                }
+
+                if ((outputRedirectBehaviour & OutputRedirectBehaviour.RedirectStdOut) !=
+                    OutputRedirectBehaviour.None)
+                {
+                    Debug.Log(outputString);
+                }
+
+                if (outputLog != null)
+                {
+                    lock (outputLog)
                     {
-                        processStandardOutput.Add(output);
+                        outputLog.AppendLine(ProcessSpatialOutput(outputString));
                     }
-                });
-                AddErrorProcessing(error =>
+                }
+
+                foreach (var outputProcessor in outputProcessors)
                 {
-                    lock (processStandardOutput)
+                    try
                     {
-                        processStandardError.Add(error);
+                        outputProcessor.Invoke(outputString);
```

Comment:
It should be noted in the doc-strings for registering processors that they'll be invoked on a different thread than the registering/main thread.

Reply:
Yeah that's a good shout